openapi.yaml

```diff
@@ -9259,3 +9341,149 @@ components:
       - account
       - work_type
       type: object
+    DeploymentMetricID:
+      description: |
+        Supported deployment metrics.
+        * `dep-count` - number of deployments in the time period.
+        * `dep-duration` - average deployment procedure time - the difference between `date_finished` and `date_started` of `DeploymentNotification`.
+        * `dep-success-count` - number of successful deployments in the time period.
+        * `dep-failure-count` - number of failed deployments in the time period.
+        * `dep-success-ratio` - ratio between successful and all deployments.
+        * `dep-size-prs` - average number of deployed pull requests.
+        * `dep-size-releases` - average number of deployed releases.
+        * `dep-size-lines` - average number of deployed lines of code (changed = insertions + removals).
+        * `dep-size-commits` - average number of deployed commits.
+        * `dep-prs-count` - number of pull requests deployed in the time period.
+        * `dep-releases-count` - number of releases deployed in the time period.
+        * `dep-lines-count` - number of deployed lines of code (changed = insertions + removals).
+        * `dep-commits-count` - number of deployed commits.
+        * `dep-repository-frequency` - average number of times a repository was deployed.
+        * `dep-jira-issues-count` - number of JIRA issues mapped to deployed pull requests.
+        * `dep-jira-bug-fixes-count` - number of deployed bug fixes.
```

Comment:
Nice metrics even though it would be more scalable to our customer's jira workflow to use the `dep-jira-issues-count` metric with some jira issue_types filter as we do in the forSet of `metrics/pull_requests`.

Reply:
OK, removing this one then.

Reply:
This is complex... The JIRA filter chooses the deployments to load, not the JIRA issues in the filtered deployments. We can add a special issue filter like `loaded_jira` in the `ForSet`.

So our original idea of looking at the bugs doesn't work because each client has their own set of labels to mark bugs?

Reply:
> So our original idea of looking at the bugs doesn't work because each client has their own set of labels to mark bugs?

No, you're right, this is working pretty fine. It's not perfect as people sometimes do their bug fixing in issue types like Flaw, Incident, Fault, ... etc but this is not very common, so relying on `dep-jira-bug-fixes-count` is going to do the job in the short term 💯